pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -338,6 +346,24 @@ def _call_args(
 
         return args, args_dict
 
+    @contextmanager
+    def _run_span(self, args_dict: dict[str, Any], tracer: Tracer) -> Iterator[None]:
+        """Create a span around the tool function call.
+
+        If installed, we want to use logfire for the span so it can take care of formatting arguments nicely
+        otherwise we call back to using opentelemetry directly.
+        """
+        span_attributes: dict[str, str | float | int] = {'pydantic_ai.tool': self.name}
```

Comment:
probably want to use https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-spans/#execute-tool-span